### ISSUE TITLE
Compatibility issue with Laravel 10

### ISSUE THREAD

**Issue by fh32000** [rollbar/rollbar-php-laravel#141]
I'm trying to use this package in my Laravel 10 project, but I'm encountering an error related to compatibility. I believe there is a conflict between the requirements of package and Laravel 10. Can you please help me resolve this issue?

```
{
    "require": {
        "php": "^8.1",
        "laravel/framework": "^10.0",
        "rollbar/rollbar-laravel": "v7.2.0"
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}


```

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - illuminate/support[v6.0.0, ..., v6.19.1] require php ^7.2 -> your php version (8.1.13) does not satisfy that requirement.
    - illuminate/support[v7.0.0, ..., v7.28.4] require php ^7.2.5 -> your php version (8.1.13) does not satisfy that requirement.
    - illuminate/support[v8.0.0, ..., v8.11.2] require php ^7.3 -> your php version (8.1.13) does not satisfy that requirement.
    - Root composer.json requires laravel/framework ^10.0 -> satisfiable by laravel/framework[10.x-dev].
    - rollbar/rollbar-laravel v7.2.0 requires illuminate/support ^6.0|^7.0|^8.0|^9.0 -> satisfiable by illuminate/support[v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev].
    - Only one of these can be installed: illuminate/support[v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev], laravel/framework[10.x-dev]. laravel/framework replaces illuminate/support and thus cannot coexist with it.
    - Root composer.json requires rollbar/rollbar-laravel v7.2.0 -> satisfiable by rollbar/rollbar-laravel[v7.2.0].

```

**Comment by jonnott** [comment]
See https://github.com/rollbar/rollbar-php-laravel/pull/140

**Comment by danielmorell** [comment]
This should be resolved by @jonnott in #140.

**Comment by jonnott** [comment]
Awaiting further PR from @michaelscola subsequent to https://github.com/rollbar/rollbar-php-laravel/pull/140#issuecomment-1431557465

I think this will need a new major version of rollbar-php also, supporting monolog v3. i.e. both this package and rollbar-php will need new major versions with a breaking change in order to support monolog v3 (and therefore Laravel 10, which requires monolog v3). 

Ironically, the Laravel 10 Upgrade Guide considers the change to requiring monolog v3 'low' impact - not if you use an external logging service via their API!

**Comment by doncullen** [comment]
Is there an estimate on when a release will be tagged? I'd love to upgrade to v10. No pressure.

**Comment by danielmorell** [comment]
If possible, I would like to get a beta out on Monday the 20th or 27th.

**Comment by jonnott** [comment]
BugSnag seem to be one step ahead on this: https://github.com/bugsnag/bugsnag-laravel/releases/tag/v2.26.0

**Comment by dinandmentink** [comment]
Just chiming in here. It seems that `dev-master` cannot be installed by composer because laravel v10 requires monolog v3.

https://laravel.com/docs/10.x/upgrade#monolog-3

**Comment by jonnott** [comment]
> Just chiming in here. It seems that `dev-master` cannot be installed by composer because laravel v10 requires monolog v3.

This is correct. `dev-master` is not ready for Laravel 10.x until `rollbar-php` version 8 is released, and then this lib is update to use that. Only then can a new version of this lib be tagged, which will need to be a non-BC version supporting Laravel 10.x, PHP 8.1, rollbar-php 8, and Monolog 3 only.

@danielmorell is on the case!

**Comment by dinandmentink** [comment]
Ah. Thanks for the explanation and headsup.

**Comment by danielmorell** [comment]
This has been resolved by #142. So `dev-master` should not work with Laravel 10. I will work on updating our CI tests and getting a beta for v8 out.

**Comment by jonnott** [comment]
> This has been resolved by #142. So `dev-master` should not work with Laravel 10. I will work on updating our CI tests and getting a beta for v8 out.

Awesome work @danielmorell and @michaelscola - just upgrading my app to Laravel 10.x now ;)
